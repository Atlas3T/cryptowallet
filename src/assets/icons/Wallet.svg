<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
