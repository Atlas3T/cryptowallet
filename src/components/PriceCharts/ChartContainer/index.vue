<template>
  <div class="chart-container">
    <div class="row justify-center">
      <q-btn
        class="chart-button"
        color="secondary"
        label="24H"
        @click="onClick(0)"
      />
      <q-btn
        class="chart-button"
        color="secondary"
        label="1W"
        @click="onClick(1)"
      />
      <q-btn
        class="chart-button"
        color="secondary"
        label="1M"
        @click="onClick(2)"
      />
    </div>

    <Chart
      v-if="chartData"
      :chart-data="chartData"
      :options="options"
      @legendHtml="updateLegend"
    />
  </div>
</template>

<script>
import Chart from '@/components/PriceCharts/Chart';

export default {
  name: 'ChartContainer',
  components: { Chart },
  props: {
    datasets: {
      type: Array,
      required: true,
    },
  },
  data() {
    return {
      newChart: '',
    };
  },
  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    chartData() {
      return {
        datasets: [{
          label: '24H',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          data: this.datasets[0].data,
        },
        {
          label: '1W',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          hidden: true,
          data: this.datasets[1].data,
        },
        {
          label: '1M',
          borderColor: '#fabc57',
          borderWidth: 3,
          pointRadius: 0,
          backgroundColor: 'black',
          hidden: true,
          data: this.datasets[2].data,
        }],
      };
    },
    options() {
      return {
        animations: {
          duration: 2000,
        },
        layout: {
          padding: {
            top: 10,
          },
        },
        scales: {
          xAxes: [{
            type: 'time',
            ticks: {
              minRotation: '45',
              fontColor: '#78d2e6',
              padding: 15,

            },
            gridLines: {
              display: false,
            },
          }],
          yAxes: [{
            position: 'left',
            color: '#78d2e6',
            ticks: {
              fontColor: '#78d2e6',
              callback: (value) => { return `${this.selectedCurrency.symbol}${value}`; },
            },
          }],
        },
        legend: {
          display: false,
        },

      };
    },
  },
  methods: {
    updateLegend(chart) {
      this.newChart = chart;
    },

    onClick(index) {
      const chart = this.newChart;
      [0, 1, 2].forEach((i) => {
        if (i !== index) {
          chart.getDatasetMeta(i).hidden = true;
        }
      });
      chart.getDatasetMeta(index).hidden = false;
      chart.update();
      this.$emit('update', index);
    },
  },
};
</script>

<style>
.chart-button{
    margin-right: 10px;
    margin-left: 10px;
    margin-bottom: 15px;
    margin-top: 10px;
    min-height: 1.75em;
}
.chart-container{
    margin: 0;
}

</style>
