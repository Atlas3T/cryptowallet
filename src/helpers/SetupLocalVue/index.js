import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
