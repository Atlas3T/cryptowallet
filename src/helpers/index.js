import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
