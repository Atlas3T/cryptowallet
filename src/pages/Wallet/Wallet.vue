<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
