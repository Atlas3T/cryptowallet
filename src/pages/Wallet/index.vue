<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
