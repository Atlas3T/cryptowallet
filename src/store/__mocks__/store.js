import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
