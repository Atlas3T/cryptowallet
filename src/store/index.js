/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
