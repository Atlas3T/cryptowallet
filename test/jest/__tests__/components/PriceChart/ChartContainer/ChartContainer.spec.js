/* eslint-disable no-magic-numbers */
import ChartContainer from '@/components/PriceCharts/ChartContainer';
import { mount } from '@vue/test-utils';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

import Wallet from '@/store/wallet/entities/wallet';
import Prices from '@/store/prices';
import Coin from '@/store/wallet/entities/coin';

const bitcoinWalletData = JSON.parse('{"$id":5,"id":5,"account_id":1,"name":"Bitcoin","displayName":"Bitcoin","symbol":"BTC","sdk":"Bitcoin","network":"BITCOIN_TESTNET","internalChainAddressIndex":0,"externalChainAddressIndex":1,"externalAddress":null,"confirmedBalance":0,"unconfirmedBalance":0,"imported":true,"enabled":true,"lastBlockHeight":0,"parentName":"","parentSdk":"","contractAddress":"","decimals":"","hdWallet":{"bip":49,"ext":{"xpriv":"xprv9zZZrtcYB6bg2sxhoVYYKAM8duZHP1MwV1SYNVLStU6qpY6NCKfuTCeeHe1MsBeYDt7XipqZYtshoGC9ghNZ6EVwugKxUPM7m2tFmExu2mt","xpub":"xpub6DYvGQ9S1U9yFN3AuX5YgJHsBwPmnU5nrEN9Ask4SodphLRWjrz9zzy88v6HFPGJuDoMg7hhEG2yFFhU7Djqx3AoE57hEL748iFp26JLXtC"},"int":{"xpriv":"xprv9zZZrtcYB6bg6LXLr8kweVuqCWUTc5dnFDJ526bEyz5pLHdfHRx2D1UPLE8bZy6hK8f1YpXQHeGF2FWEds5S1MSbTH2boaQRNUoz6vy4MRq","xpub":"xpub6DYvGQ9S1U9yJpboxAHx1drZkYJx1YMdcSDfpUzrYKcoD5xopyGGkonsBXbmCnHteYYMcjgWRU7EyByaGf8Mav4y5BTkYmC2TD2JVPmkJxs"},"type":1,"network":{"name":"BITCOIN_TESTNET","type":"testnet","bip":1,"segwit":true,"discovery":"http://92.207.178.198:3001/api","broadcastUrl":"https://chain.so/api/v2/send_tx/BTCTEST","feeApi":"https://api.blockcypher.com/v1/btc/main","connect":{"messagePrefix":"Bitcoin Signed Message:","bech32":"tb","bip32":{"public":70617039,"private":70615956},"pubKeyHash":111,"scriptHash":196,"wif":239}}},"erc20Wallet":""}');
const bitcoinData = JSON.parse('{"$id":"Bitcoin","name":"Bitcoin","displayName":"Bitcoin","minConfirmations":6,"sdk":"Bitcoin","symbol":"BTC","network":"BITCOIN_TESTNET","denomination":"0.00000000","parentName":"","parentSdk":"","contractAddress":"","decimals":""}');
const chartDataDay = JSON.parse('{"$id":"BTC_GBP_day","coin":"BTC","currency":"GBP","period":"day","updated":1555516399799,"data":[{"t":1555426800000,"y":3907.54},{"t":1555430400000,"y":3907.63},{"t":1555434000000,"y":3987.8},{"t":1555437600000,"y":3981.87},{"t":1555441200000,"y":3990.52},{"t":1555444800000,"y":3978.55},{"t":1555448400000,"y":3985.98},{"t":1555452000000,"y":3997},{"t":1555455600000,"y":3992.39},{"t":1555459200000,"y":4023.61},{"t":1555462800000,"y":4010.39},{"t":1555466400000,"y":4004.59},{"t":1555470000000,"y":4007.47},{"t":1555473600000,"y":4010.36},{"t":1555477200000,"y":4000.12},{"t":1555480800000,"y":3994.68},{"t":1555484400000,"y":4000.15},{"t":1555488000000,"y":4001.41},{"t":1555491600000,"y":3993.82},{"t":1555495200000,"y":4017.2},{"t":1555498800000,"y":4028.62},{"t":1555502400000,"y":3996.31},{"t":1555506000000,"y":3997.83},{"t":1555509600000,"y":4004.72},{"t":1555513200000,"y":3998.87}]}');
const chartDataWeek = JSON.parse('{"$id":"BTC_GBP_week","coin":"BTC","currency":"GBP","period":"week","updated":1555516399800,"data":[{"t":1554908400000,"y":4033.13},{"t":1554912000000,"y":4028.07},{"t":1554915600000,"y":4048.69},{"t":1554919200000,"y":4144.66},{"t":1554922800000,"y":4114.58},{"t":1554926400000,"y":4061.61},{"t":1554930000000,"y":4060},{"t":1554933600000,"y":4048.93},{"t":1554937200000,"y":4071.02},{"t":1554940800000,"y":4056.75},{"t":1554944400000,"y":4052.23},{"t":1554948000000,"y":4050.25},{"t":1554951600000,"y":3971.01},{"t":1554955200000,"y":3984.56},{"t":1554958800000,"y":3990.75},{"t":1554962400000,"y":3992.36},{"t":1554966000000,"y":3979.87},{"t":1554969600000,"y":3983.12},{"t":1554973200000,"y":3912.31},{"t":1554976800000,"y":3880.79},{"t":1554980400000,"y":3885.26},{"t":1554984000000,"y":3883.42},{"t":1554987600000,"y":3873.52},{"t":1554991200000,"y":3868.3},{"t":1554994800000,"y":3873.78},{"t":1554998400000,"y":3873.48},{"t":1555002000000,"y":3875.95},{"t":1555005600000,"y":3903.9},{"t":1555009200000,"y":3892.73},{"t":1555012800000,"y":3876.06},{"t":1555016400000,"y":3861.58},{"t":1555020000000,"y":3881.68},{"t":1555023600000,"y":3876.32},{"t":1555027200000,"y":3864.79},{"t":1555030800000,"y":3819.55},{"t":1555034400000,"y":3839.41},{"t":1555038000000,"y":3849.68},{"t":1555041600000,"y":3845.91},{"t":1555045200000,"y":3845.45},{"t":1555048800000,"y":3856.86},{"t":1555052400000,"y":3856.34},{"t":1555056000000,"y":3853.31},{"t":1555059600000,"y":3860.26},{"t":1555063200000,"y":3895.86},{"t":1555066800000,"y":3881.11},{"t":1555070400000,"y":3886.38},{"t":1555074000000,"y":3871.65},{"t":1555077600000,"y":3884.84},{"t":1555081200000,"y":3881.01},{"t":1555084800000,"y":3894.17},{"t":1555088400000,"y":3892.76},{"t":1555092000000,"y":3887.62},{"t":1555095600000,"y":3880.11},{"t":1555099200000,"y":3875.94},{"t":1555102800000,"y":3884.32},{"t":1555106400000,"y":3894.22},{"t":1555110000000,"y":3891.61},{"t":1555113600000,"y":3884.92},{"t":1555117200000,"y":3874.34},{"t":1555120800000,"y":3903.48},{"t":1555124400000,"y":3919.73},{"t":1555128000000,"y":3922.75},{"t":1555131600000,"y":3920.06},{"t":1555135200000,"y":3914.46},{"t":1555138800000,"y":3902.48},{"t":1555142400000,"y":3891.67},{"t":1555146000000,"y":3888.71},{"t":1555149600000,"y":3873.67},{"t":1555153200000,"y":3884.35},{"t":1555156800000,"y":3898.09},{"t":1555160400000,"y":3893.36},{"t":1555164000000,"y":3915.08},{"t":1555167600000,"y":3904.48},{"t":1555171200000,"y":3897.62},{"t":1555174800000,"y":3913.85},{"t":1555178400000,"y":3909.69},{"t":1555182000000,"y":3899.17},{"t":1555185600000,"y":3905.56},{"t":1555189200000,"y":3898.75},{"t":1555192800000,"y":3893.58},{"t":1555196400000,"y":3895.23},{"t":1555200000000,"y":3890.86},{"t":1555203600000,"y":3891.6},{"t":1555207200000,"y":3887.94},{"t":1555210800000,"y":3885.81},{"t":1555214400000,"y":3882.76},{"t":1555218000000,"y":3865.61},{"t":1555221600000,"y":3869.94},{"t":1555225200000,"y":3893.21},{"t":1555228800000,"y":3895.73},{"t":1555232400000,"y":3899.46},{"t":1555236000000,"y":3887.73},{"t":1555239600000,"y":3896.83},{"t":1555243200000,"y":3896.18},{"t":1555246800000,"y":3896.74},{"t":1555250400000,"y":3901.94},{"t":1555254000000,"y":3902.81},{"t":1555257600000,"y":3907.69},{"t":1555261200000,"y":3909.22},{"t":1555264800000,"y":3906.1},{"t":1555268400000,"y":3908.29},{"t":1555272000000,"y":3910.29},{"t":1555275600000,"y":3917.14},{"t":1555279200000,"y":3949.01},{"t":1555282800000,"y":3949.76},{"t":1555286400000,"y":3952.87},{"t":1555290000000,"y":3955.66},{"t":1555293600000,"y":3957.5},{"t":1555297200000,"y":3943.74},{"t":1555300800000,"y":3962.51},{"t":1555304400000,"y":3963.71},{"t":1555308000000,"y":3954.8},{"t":1555311600000,"y":3945.37},{"t":1555315200000,"y":3941.02},{"t":1555318800000,"y":3935.87},{"t":1555322400000,"y":3934.41},{"t":1555326000000,"y":3926.78},{"t":1555329600000,"y":3935.06},{"t":1555333200000,"y":3929.9},{"t":1555336800000,"y":3929.48},{"t":1555340400000,"y":3938.79},{"t":1555344000000,"y":3918.15},{"t":1555347600000,"y":3881.85},{"t":1555351200000,"y":3868.6},{"t":1555354800000,"y":3862.25},{"t":1555358400000,"y":3846},{"t":1555362000000,"y":3861.69},{"t":1555365600000,"y":3861.44},{"t":1555369200000,"y":3855.48},{"t":1555372800000,"y":3868.28},{"t":1555376400000,"y":3871.25},{"t":1555380000000,"y":3857.04},{"t":1555383600000,"y":3864.16},{"t":1555387200000,"y":3876.04},{"t":1555390800000,"y":3887.71},{"t":1555394400000,"y":3879.18},{"t":1555398000000,"y":3893.2},{"t":1555401600000,"y":3888.82},{"t":1555405200000,"y":3891.73},{"t":1555408800000,"y":3877.79},{"t":1555412400000,"y":3882.53},{"t":1555416000000,"y":3864.49},{"t":1555419600000,"y":3895.67},{"t":1555423200000,"y":3886.25},{"t":1555426800000,"y":3907.54},{"t":1555430400000,"y":3907.63},{"t":1555434000000,"y":3987.8},{"t":1555437600000,"y":3981.87},{"t":1555441200000,"y":3990.52},{"t":1555444800000,"y":3978.55},{"t":1555448400000,"y":3985.98},{"t":1555452000000,"y":3997},{"t":1555455600000,"y":3992.39},{"t":1555459200000,"y":4023.61},{"t":1555462800000,"y":4010.39},{"t":1555466400000,"y":4004.59},{"t":1555470000000,"y":4007.47},{"t":1555473600000,"y":4010.36},{"t":1555477200000,"y":4000.12},{"t":1555480800000,"y":3994.68},{"t":1555484400000,"y":4000.15},{"t":1555488000000,"y":4001.41},{"t":1555491600000,"y":3993.82},{"t":1555495200000,"y":4017.2},{"t":1555498800000,"y":4028.62},{"t":1555502400000,"y":3996.31},{"t":1555506000000,"y":3997.83},{"t":1555509600000,"y":4004.72},{"t":1555513200000,"y":3998.87}]}');
const chartDataMonth = JSON.parse('{"$id":"BTC_GBP_month","coin":"BTC","currency":"GBP","period":"month","updated":1555516399801,"data":[{"t":1552780800000,"y":2995.29},{"t":1552867200000,"y":3008.11},{"t":1552953600000,"y":3023.25},{"t":1553040000000,"y":3067.08},{"t":1553126400000,"y":3048.32},{"t":1553212800000,"y":3031.65},{"t":1553299200000,"y":3033.26},{"t":1553385600000,"y":3012.14},{"t":1553472000000,"y":2981.16},{"t":1553558400000,"y":2975.35},{"t":1553644800000,"y":3060.8},{"t":1553731200000,"y":3078.33},{"t":1553817600000,"y":3150.97},{"t":1553904000000,"y":3162.44},{"t":1553990400000,"y":3148.41},{"t":1554076800000,"y":3179.54},{"t":1554163200000,"y":3727.96},{"t":1554249600000,"y":3796.51},{"t":1554336000000,"y":3759.98},{"t":1554422400000,"y":3864.58},{"t":1554508800000,"y":3892.08},{"t":1554595200000,"y":3977.44},{"t":1554681600000,"y":4063.33},{"t":1554768000000,"y":3976.55},{"t":1554854400000,"y":4071.02},{"t":1554940800000,"y":3876.32},{"t":1555027200000,"y":3891.61},{"t":1555113600000,"y":3895.23},{"t":1555200000000,"y":3949.76},{"t":1555286400000,"y":3855.48},{"t":1555372800000,"y":3992.39},{"t":1555459200000,"y":3999.51}]}');

describe('components/ChartContainer', () => {
  let storeMocks;
  let wrapper;
  let router;

  const mockDatasets = [{ hidden: false }, { hidden: false }, { hidden: false }];

  const mockChart = {
    getDatasetMeta: jest.fn().mockImplementation((i) => {
      return mockDatasets[i];
    }),
    update: jest.fn(),
  };

  function wrapperInit(options) {
    return mount(ChartContainer, options);
  }

  function storeInit(custom) {
    storeMocks = createStoreMocks(custom);
    Wallet.insert({ data: [bitcoinWalletData] });
    Coin.insert({ data: [bitcoinData] });
    Prices.insert({ data: [chartDataDay, chartDataWeek, chartDataMonth] });

    router = createRouter(storeMocks.store);
    router.push({ path: '/wallet/single/prices/5' });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      store: storeMocks.store,
      stubs: { Chart: true },
      propsData: { datasets: [chartDataDay, chartDataWeek, chartDataMonth] },
    });
  }
  beforeEach(() => {
    jest.clearAllMocks();
    return storeInit();
  });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders three time period buttons', () => {
    expect(wrapper.findAll('.chart-button').length).toEqual(3);
  });

  it('changes the chart data when a time period button is clicked', () => {
    wrapper.vm.updateLegend(mockChart);
    wrapper.findAll('.chart-button').at(1).trigger('click');
    expect(mockChart.getDatasetMeta).toHaveBeenCalledTimes(3);
    expect(mockDatasets[0].hidden).toBe(true);
    expect(mockDatasets[2].hidden).toBe(true);
    expect(mockDatasets[1].hidden).toBe(false);

    wrapper.findAll('.chart-button').at(2).trigger('click');
    expect(mockDatasets[0].hidden).toBe(true);
    expect(mockDatasets[2].hidden).toBe(false);
    expect(mockDatasets[1].hidden).toBe(true);

    wrapper.findAll('.chart-button').at(0).trigger('click');
    expect(mockDatasets[0].hidden).toBe(false);
    expect(mockDatasets[2].hidden).toBe(true);
    expect(mockDatasets[1].hidden).toBe(true);
  });

  it('displays the slected currency symbol on the chart y axis', () => {
    const yAxisValue = wrapper.vm.options.scales.yAxes[0].ticks.callback(5);
    expect(yAxisValue).toEqual('£5');
  });
});
