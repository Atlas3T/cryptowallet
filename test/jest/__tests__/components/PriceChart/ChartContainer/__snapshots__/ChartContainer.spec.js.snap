// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChartContainer renders and matches snapshot 1`] = `
<div
  class="chart-container"
>
  <div
    class="row justify-center"
  >
    <button
      class="q-btn inline q-btn-item non-selectable chart-button q-btn--standard q-btn--rectangle bg-secondary text-white q-focusable q-hoverable"
      tabindex="0"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <div>
          24H
        </div>
      </div>
    </button>
     
    <button
      class="q-btn inline q-btn-item non-selectable chart-button q-btn--standard q-btn--rectangle bg-secondary text-white q-focusable q-hoverable"
      tabindex="0"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <div>
          1W
        </div>
      </div>
    </button>
     
    <button
      class="q-btn inline q-btn-item non-selectable chart-button q-btn--standard q-btn--rectangle bg-secondary text-white q-focusable q-hoverable"
      tabindex="0"
      type="button"
    >
      <div
        class="q-focus-helper"
        tabindex="-1"
      />
      <div
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <div>
          1M
        </div>
      </div>
    </button>
  </div>
   
  <chart-stub
    chartdata="[object Object]"
    chartid="line-chart"
    cssclasses=""
    height="400"
    options="[object Object]"
    plugins=""
    width="400"
  />
</div>
`;
